src/main/java/de/tum/in/www1/artemis/repository/ComplaintRepository.java

```diff
@@ -24,6 +24,9 @@
 @Repository
 public interface ComplaintRepository extends JpaRepository<Complaint, Long> {
 
+    @Query("SELECT c FROM Complaint c LEFT JOIN c.result r LEFT JOIN r.submission s WHERE s.id = :#{#submissionId}")
+    Optional<Complaint> findByResultSubmissionId(Long submissionId);
```

Comment:
Please add the `@Param` annotation here
```suggestion
    Optional<Complaint> findByResultSubmissionId(@Param("submissionId") Long submissionId);
```
Maybe you could also just rename it to `findBySubmissionId` I guess